examples/ai-image-generator/pages/index.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from "react";
+import initMiro from "../initMiro";
+//React components for the prompt and button
+import PromptInput from "../components/PromptInput";
+import Button from "../components/Button";
+
+export const getServerSideProps = async function getServerSideProps({ req }) {
+  const { miro } = initMiro(req);
+  // redirect to auth url if user has not authorized the app
+  if (!(await miro.isAuthorized(""))) {
+    return {
+      props: {
+        authUrl: miro.getAuthUrl(),
+      },
+    };
+  }
+  return {
+    props: {},
+  };
+};
+
+export default function Main({ boards, authUrl }) {
+  useEffect(() => {
+    removeSpinner();
+    // Opens the panel for our app UI when we click on icon in the left sidebar
+    if (new URLSearchParams(window.location.search).has("panel")) return;
+    window.miro.board.ui.on("icon:click", async () => {
+      window.miro.board.ui.openPanel({
+        url: `/?panel=1`,
+      });
+    });
+  }, []);
+
+  // Register the drop event handler once.
+  useEffect(() => {
+    window.miro.board.ui.on("drop", drop);
+  }, []);
+
+  //drag and drop logic
+  const drop = async (e) => {
+    showSpinner();
+    const { x, y, target } = e;
+    if (target instanceof HTMLImageElement) {
+      const image = await window.miro.board.createImage({
+        x,
+        y,
+        url: target.src,
+      });
+      await window.miro.board.viewport.zoomTo(image);
+    }
+    removeSpinner();
+  };
+
+  const [inputValue, setInputValue] = useState("");
+
+  //handles the prompt input being typed in
+  const handleInputChange = (newValue) => {
+    setInputValue(newValue);
+  };
+
+  // Shows spinner while API calls are in progress / image is being dragged & dropped
+  const showSpinner = async () => {
+    let spinner = await document.getElementById("spinner");
```

Comment:
`getElementById` is the sync method:

```suggestion
  const showSpinner = () => {
    const spinner = document.getElementById("spinner");
```

Reply:
Thank you - just changed